### ISSUE TITLE
Automatically downloads contents of onedrive folder 

### ISSUE THREAD

**Issue by aaronsuydam** [files-community/Files#14224]
### Description

I store my downloads folder in OneDrive. When I navigate to the folder using Files, it opens and then begins to download all the files in my downloads that have been removed from my device to save space (the "files-on-demand" functionality of OneDrive). This poses a major problem because I have very large files that I download, to keep them all on my device is not possible. I'd be happy to provide logs or something if someone wants to tell me where to look and what sections would be helpful. 

### Steps To Reproduce

1. Change download folder's location to a path in OneDrive.
2. Populate that folder and then allow OneDrive to automatically free up space.
3. Navigate to the folder using Files. 

### Requirements

The folder should open and be fully interactable (sort by date/name, scroll, etc) without downloading any files.

### Files Version

3.0.15.0

### Windows Version

10.0.22621.2715

### Log File

[debug.log](https://github.com/files-community/Files/files/13637744/debug.log)

**Comment by yaira2** [comment]
Can you also clarify
1. Is files on demand turned on or off?
2. Are there any exe files in this location?
3. Does this happen for every location, or only some locations?
4. Are the files fully online, or are they partially synced?

**Comment by NightmareZero** [comment]
> Can you also clarify
> 
> 1. Is files on demand turned on or off?
> 2. Are there any exe files in this location?
> 3. Does this happen for every location, or only some locations?
> 4. Are the files fully online, or are they partially synced?

I encountered the same problem.
1. on demand is on
2. no exe in this location
3. in some localtion
4.  partially synced, and there is always a re-download some files when I perform 'free up space'.

**Comment by yaira2** [comment]
@NightmareZero which version of Files are you using? Can you confirm if this happens in all OneDrive locations or is it only some of them?

**Comment by NightmareZero** [comment]
> @NightmareZero which version of Files are you using? Can you confirm if this happens in all OneDrive locations or is it only some of them? 

If that directory has an exe file, it will occur at any location
If there is no exe file, it will occur in certain specific directories, but I have not found any pattern

The latest version I am using is from the Windows App Store, and the version is 3.0.15.0， windows version 10.0.22631.2861

**Comment by yaira2** [comment]
Are there any folders in those locations?

**Comment by aaronsuydam** [comment]
> Can you also clarify
> 
>     1. Is files on demand turned on or off?
> 
>     2. Are there any exe files in this location?
> 
>     3. Does this happen for every location, or only some locations?
> 
>     4. Are the files fully online, or are they partially synced?

Hey @yaira2 im so sorry, with finals and everything this got lost. 

In order, 0. No i do not have calculate folder sizes turned on.

1. Files on Demand is turned on.
2. Yes, many. It is my cloud repository of software packages. I keep the folder in onedrive so that if i run into a situation where i have to re download an obscure piece of software (as happens from time to time), I don't suffer through a server that has the bandwidth of yesteryear.
3. It happens specifically with this location. 
4. Some files are online, some are on device. It only starts downloading the ones that are not yet on device.

**Comment by NightmareZero** [comment]
> Are there any folders in those locations?

Yes, there are some other directories

**Comment by yaira2** [comment]
Does anyone experiencing this issue have Visual Studio on their device? I know it's a lot to ask but it'll help if one of you can build from the source code and place some break points to help us debug.

**Comment by mbartlett21** [comment]
It happens for me when I have 'Show alternate data streams' turned on. Turning off stops it downloading.
(Under Settings > Folders > Hidden items)

**Comment by yaira2** [comment]
@aaronsuydam @NightmareZero can you confirm if that's the case for you as well?

**Comment by yaira2** [comment]
@mbartlett21 would you be interested in working on a fix?

**Comment by aaronsuydam** [comment]
@yaira2 it is not the case for me. Just checked.

**Comment by mbartlett21** [comment]
> @mbartlett21 would you be interested in working on a fix?

@yaira2 
Fix for alternate data streams done, tested and PR: https://github.com/files-community/Files/pull/14398

**Comment by mbartlett21** [comment]
@aaronsuydam 
Do the files that are automatically downloading for you have tags assigned in Files?

**Comment by mbartlett21** [comment]
@aaronsuydam and @NightmareZero 
Would you be able to post your settings here?

1. Click on Settings > Advanced > Export settings
2. Save the zip file.
    (note: this zip file includes a database of your assigned tags, so you probably don't want to post that directly)
3. Extract the user_settings.json file out of it and post that.

**Comment by NightmareZero** [comment]
> @aaronsuydam and @NightmareZero Would you be able to post your settings here?
> 
> 1. Click on Settings > Advanced > Export settings
> 2. Save the zip file.
>    (note: this zip file includes a database of your assigned tags, so you probably don't want to post that directly)
> 3. Extract the user_settings.json file out of it and post that.

```json
{
  "ShowSendToMenu": true,
  "ShowOpenInNewWindow": true,
  "AppThemeSidebarBackgroundColor": "",
  "ShowGitLastCommitDateColumn": false,
  "ShowOpenInNewPane": true,
  "DateTimeFormat": 0,
  "NameColumnWidth": 240,
  "UseCompactStyles": false,
  "GitStatusColumnWidth": 80,
  "OpenItemsWithOneClick": false,
  "AppThemeBackgroundColor": "#00000000",
  "SelectFilesOnHover": false,
  "GitCommitAuthorColumnWidth": 140,
  "AppThemeFontFamily": "Segoe UI Variable",
  "DoubleClickToGoUp": true,
  "DefaultSortDirectoriesAlongsideFiles": false,
  "ShowNetworkDrivesSection": false,
  "ShowQuickAccessWidget": true,
  "ShowDateColumn": true,
  "IsSidebarOpen": true,
  "RestoreTabsOnStartup": false,
  "AppThemeAddressBarBackgroundColor": "",
  "ShowGitLastCommitShaColumn": false,
  "ShowFileExtensionWarning": true,
  "DefaultDirectorySortDirection": 0,
  "ShowGitLastCommitMessageColumn": false,
  "ColumnLayoutOpenFoldersWithOneClick": true,
  "AlwaysOpenDualPaneInNewTab": false,
  "ShowSyncStatusColumn": true,
  "DefaultLayoutMode": 4,
  "GitLastCommitMessageColumnWidth": 140,
  "Actions": null,
  "ContinueLastSessionOnStartUp": true,
  "ShowFavoritesSection": true,
  "TypeColumnWidth": 140,
  "ShowStatusCenterTeachingTip": false,
  "ShowCheckboxesWhenSelectingItems": true,
  "ShowGitStatusColumn": false,
  "ShowOpenInNewTab": true,
  "ShowDrivesSection": true,
  "ShowSizeColumn": true,
  "ShowFileExtensions": true,
  "SyncStatusColumnWidth": 50,
  "DefaultGridViewSize": 160,
  "MoveShellExtensionsToSubMenu": true,
  "SidebarWidth": 255,
  "ShowTypeColumn": true,
  "GitLastCommitDateColumnWidth": 140,
  "DateModifiedColumnWidth": 200,
  "ClickedToReviewApp": false,
  "SearchUnindexedItems": false,
  "ShowProtectedSystemFiles": false,
  "ShowGitCommitAuthorColumn": false,
  "ShowThumbnails": true,
  "OpenSpecificPageOnStartup": false,
  "ShowHiddenItems": false,
  "ShowFileTagsWidget": false,
  "IsEnabled": false,
  "ShowDateDeletedColumn": true,
  "AreAlternateStreamsVisible": false,
  "LeaveAppRunning": true,
  "ConflictsResolveOption": 0,
  "ShowDotFiles": true,
  "RecentFilesWidgetExpanded": true,
  "ShowRecentFilesWidget": true,
  "ShowDateCreatedColumn": false,
  "OpenNewTabOnStartup": false,
  "ShowFileTagColumn": true,
  "DateCreatedColumnWidth": 200,
  "ShowPathColumn": true,
  "DeleteConfirmationPolicy": 0,
  "DrivesWidgetExpanded": true,
  "SizeColumnWidth": 100,
  "TabsOnStartupList": null,
  "ShowFileTagsSection": true,
  "OpenFoldersInNewTab": false,
  "SyncFolderPreferencesAcrossDirectories": false,
  "AppThemeBackdropMaterial": 2,
  "CalculateFolderSizes": false,
  "DefaultSortOption": 0,
  "ShowOriginalPathColumn": true,
  "ShowCloudDrivesSection": true,
  "ShowWslSection": true,
  "ShowLibrarySection": false,
  "PathColumnWidth": 200,
  "DateDeletedColumnWidth": 200,
  "DefaultGroupByDateUnit": 0,
  "FoldersWidgetExpanded": true,
  "GitLastCommitShaColumnWidth": 80,
  "DefaultGroupOption": 0,
  "OriginalPathColumnWidth": 200,
  "ShowCompressionOptions": true,
  "ShowEditTagsMenu": true,
  "DefaultSortFilesFirst": false,
  "DefaultDirectoryGroupDirection": 0,
  "AppThemeFileAreaBackgroundColor": "",
  "TagColumnWidth": 140,
  "ShowDrivesWidget": true,
  "OpenTabInExistingInstance": true
}
```

**Comment by mbartlett21** [comment]
@NightmareZero
I don't see anything too much different from my settings which aren't causing downloads.

Do the files that are downloading for you have tags assigned? Having tags assigned will cause those files to download.

**Comment by NightmareZero** [comment]
> @NightmareZero I don't see anything too much different from my settings which aren't causing downloads.
> 
> Do the files that are downloading for you have tags assigned? Having tags assigned will cause those files to download.

no tag assigned

**Comment by yaira2** [comment]
I have a fix for the issue in #14597
